### PR TITLE
cri-tools 1.22.0 (new formula)

### PR DESCRIPTION
**Pull request by badouralix**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

https://github.com/kubernetes-sigs/cri-tools contains two tools, namely `crictl` and `critest`, that can become handy on macos to test custom container runtimes

It was requested a few months ago in https://github.com/kubernetes-sigs/cri-tools/issues/587

This formula provides this specific release https://github.com/kubernetes-sigs/cri-tools/releases/tag/v1.22.0

-----

`BINDIR` envvar is set in the formula so that `make install` installs the binary in the right place

See rule in https://github.com/kubernetes-sigs/cri-tools/blob/v1.22.0/Makefile#L90-L92

`VERSION` envvar is only necessary when there is no git repository to fetch the version from

See version inference in https://github.com/kubernetes-sigs/cri-tools/blob/v1.22.0/Makefile#L35-L37

### CODE REVIEW

File: Formula/cri-tools.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    output = Utils.safe_popen_read("#{bin}/crictl", "completion", "zsh")
+    (zsh_completion/"_crictl").write output
+
+    output = Utils.safe_popen_read("#{bin}/crictl", "completion", "fish")
+    (fish_completion/"crictl.fish").write output
+  end
+
+  test do
+    assert_match "crictl version", shell_output("#{bin}/crictl --version 2>&1")
+    assert_match "critest version", shell_output("#{bin}/critest --version 2>&1")
```
Comment (carlocab) [inline Formula/cri-tools.rb:32]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

**Review by carlocab — COMMENTED** [review]
Thanks for the PR, @badouralix. Just one comment.

File: Formula/cri-tools.rb (hunk 1 above)
Comment (badouralix) [reply Formula/cri-tools.rb]: It looks like `critest` is easy to test thanks to its dry-run mode.

On the other hand, just like any other `...ctl` tool, `crictl` is a bit more tricky. Tentatively testing if it successfully fails in https://github.com/Homebrew/homebrew-core/pull/88744/commits/c704754dbe49054d77862db020655c204fc79e16.

> If the software cannot function without credentials or requires a virtual machine, docker instance, etc. to run, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected. This is preferred over mocking a dependency.

**Review by chenrui333 — APPROVED** [review]
LGTM

**Review by carlocab — APPROVED** [review]
Thanks, @badouralix!

**Pull request closed without merging**